lib/iris/coords.py

```diff
@@ -1510,14 +1510,6 @@ def contains_point(self, point):
         """
         if self.bound is None:
             raise ValueError("Point cannot exist inside an unbounded cell.")
-        if hasattr(point, "timetuple") or np.any(
-            [hasattr(val, "timetuple") for val in self.bound]
-        ):
-            raise TypeError(
-                "Cannot determine whether a point lies within "
-                "a bounded region for datetime-like objects."
-            )
-
```

Comment:
There is a remaining similar block, which I assume needs similar changes?

https://github.com/SciTools/iris/blob/c48829aab5978ef9c60259d69b1b95cddfed3c68/lib/iris/coords.py#L1459-L1472

Reply:
Seems this would need some analogous testing as well?

Reply:
added tests to _common_cmp_ to match this